SPDX-License-Identifier: GPL-3.0-or-later

Fix list function declarations to use proper prototypes for C23.

--- bash-4.4/externs.h
+++ bash-4.4/externs.h
@@ -141,10 +141,10 @@
 /* Declarations for functions defined in list.c. */
 extern void list_walk __P((GENERIC_LIST *, sh_glist_func_t *));
 extern void wlist_walk __P((WORD_LIST *, sh_icpfunc_t *));
-extern GENERIC_LIST *list_reverse ();
-extern int list_length ();
-extern GENERIC_LIST *list_append ();
-extern GENERIC_LIST *list_remove ();
+extern GENERIC_LIST *list_reverse __P((GENERIC_LIST *));
+extern int list_length __P((GENERIC_LIST *));
+extern GENERIC_LIST *list_append __P((GENERIC_LIST *, GENERIC_LIST *));
+extern GENERIC_LIST *list_remove __P((GENERIC_LIST **, Function *, char *));
 
 /* Declarations for functions defined in stringlib.c */
 extern int find_string_in_alist __P((char *, STRING_INT_ALIST *, int));
